decentraland/unity-renderer
# feat: Highligh #channel names pattern in any text

## What does this PR change?
Every text block in the Explorer, that contains a name of a channel pattern containing #, will automatically convert to a clickable link, in the same way we do with coordinates. 
> Hey! Everybody! Join the [#wondermine](#wondermine) channel and while you play!

Clicking on it, will prompt a dialog to confirm the player wants to join that channel. This will apply to any text in the chat, places or events descriptions, or profile information.

[Documentation Link](https://www.notion.so/decentraland/PRD-Channels-open-for-comments-1a51033cc6134889a4d863d0bd1cb607#cbe51fde62074f2c8a0b30fe30ce7e65)

**Acceptance Criteria**
- Any text meeting criteria needs to be highlighted, anywhere
- Clicking on one of those links, displays a confirmation toast
- Accepting will open the conversation in the chat panel

## How to test the changes?
1. Go to: https://play.decentraland.zone/?renderer-branch=feat/highligh-channel-names-pattern-in-any-text
2. Open any chat window.
3. Write different text containing several channel patterns inside.
4. Send the text pressing ENTER.
5. Notice the channel patters in the text are automatically converted into links, and if you click on them, a new UI modal will be open and will let you go to the corresponding channel.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
